daemons/based/based_notify.c

```diff
@@ -167,19 +168,21 @@ cib_diff_notify(int options, const char *client, const char *call_id, const char
 
     if (add_updates != del_updates) {
         do_crm_log(log_level,
-                   "Update (client: %s%s%s): %d.%d.%d -> %d.%d.%d (%s)",
-                   client,
+                   "Update (client: %s%s%s): %d.%d.%d -> %d.%d.%d from %s (%s)",
+                   client_name,
                    ((call_id != NULL)? ", call:" : ""), pcmk__s(call_id, ""),
                    del_admin_epoch, del_epoch, del_updates,
                    add_admin_epoch, add_epoch, add_updates,
+                   pcmk__s(origin, "unspecified peer"),
                    pcmk_strerror(result));
 
     } else {
         do_crm_log(log_level,
-                   "Local-only change (client: %s%s%s): %d.%d.%d (%s)",
-                   client,
+                   "Local-only change (client: %s%s%s): %d.%d.%d from %s (%s)",
+                   client_name,
                    ((call_id != NULL)? ", call: " : ""), pcmk__s(call_id, ""),
                    add_admin_epoch, add_epoch, add_updates,
+                   pcmk__s(origin, "unspecified peer"),
```

Comment:
I copied the unspecified peer part from `cib_replace_notify()`, but that might be irrelevant for local-only changes, and/or for diffs.

Reply:
Try it and see what it looks like

Reply:
Seems fine, kind of:
```
Mar 30 12:44:13.704 fastvm-rhel-9-0-42 pacemaker-based     [528201] (cib_diff_notify@based_notify.c:180)        trace: Local-only change (client: cibadmin, call: 2): 0.8.0 from fastvm-rhel-9-0-42 (OK)
```

I wouldn't have expected a "local-only" change to be able to come from a peer though:
```
Mar 30 12:44:15.334 fastvm-rhel-9-0-42 pacemaker-based     [528201] (cib_process_request@based_callbacks.c:1069)        info: Completed cib_modify operation for section crm_config: OK (rc=0, origin=fastvm-rhel-9-0-43/crmd/15, version=0.8.0)
Mar 30 12:44:15.336 fastvm-rhel-9-0-42 pacemaker-based     [528201] (cib_process_replace@cib_ops.c:226)         info: Digest matched on replace from fastvm-rhel-9-0-43: 326e813cc7c50686cf09bebfdb717714
Mar 30 12:44:15.336 fastvm-rhel-9-0-42 pacemaker-based     [528201] (cib_process_replace@cib_ops.c:260)         info: Replaced 0.8.0 with 0.8.0 from fastvm-rhel-9-0-43
Mar 30 12:44:15.336 fastvm-rhel-9-0-42 pacemaker-based     [528201] (log_info@output_log.c:245)         info: Local-only Change: 0.8.0
Mar 30 12:44:15.336 fastvm-rhel-9-0-42 pacemaker-based     [528201] (log_info@output_log.c:245)         info: +  /cib:  @cib-last-written=Thu Mar 30 12:44:12 2023
Mar 30 12:44:15.337 fastvm-rhel-9-0-42 pacemaker-controld  [528206] (connect_and_send_attrd_request@ipc_attrd.c:178)    info: Connecting to cluster... 5 retries remaining
Mar 30 12:44:15.338 fastvm-rhel-9-0-42 pacemaker-based     [528201] (cib_diff_notify@based_notify.c:180)        trace: Local-only change (client: crmd, call: 18): 0.8.0 from fastvm-rhel-9-0-43 (OK)
```

So it sounds like "Local-only" might be misleading, though it doesn't necessarily need to be changed right now (partly because I don't yet know what it implies).

In any case, since the origin can be non-empty, it's worth keeping.

Reply:
"Local-only" is definitely not determined correctly here; xml_show_patchset_header() has the correct method (if any age field changed, it's a diff). Presumably "local-only" refers to the fact that some fields are exempt from digesting (origin is supposed to be one of them), so if that's the only thing that changed, there will be no diff sent around.

The "origin" field is the node where the original CIB request originated. The fact that it's a different node in this case is probably from the incorrect method of checking local-only.

Reply:
Even after making the logic match `xml_show_patchset_header()`, we can still see a "Local-only" diff from a peer:
```
Mar 30 15:40:18.857 fastvm-rhel-9-0-42 pacemaker-based     [536035] (cib_diff_notify@based_notify.c:187)        trace: Local-only change to CIB generation 0.15.0 from client crmd call 18 (fastvm-rhel-9-0-43) (OK)
```

Not gonna worry about it for now